pkg/kubectl/visibility/BUILD

```diff
@@ -0,0 +1,28 @@
+# This is the list of dependencies allowed to use kubectl.
+package_group(
+    name = "kubectl_allowed_deps",
```

Comment:
I'm a fan of getting the kubectl related package groups under the control of a sig-cli OWNERS file, and happy that util/i18n, util/crlf etc have been locked down - but not a fan of the collapse.

Those groups - pkg_kubectl_util_CONSUMERS, pkg_kubectl_resource_CONSUMERS, pkg_kubectl_cmd_util_CONSUMERS_BAD,  etc. -  with their detailed lists of packages - constitutes a visibility graph.

They represent my notes of a boring afternoon doing bazel queries and coding the results in  bazel-enforced rules.  It allows one to see by inspection where hyperkube gets involved, and what_other_ packages have a similar dependence - allowing one to scope that particular extraction problem relative to other extraction problems.   That's hard to so with just one query.

This PR takes that detailed visibility graph and collapses it to two nodes - kubectl/... and not-kubectl.  Yes, sure.   I want to collapse them too, but as a check-off list to actually doing the work to break the bad deps.  I.e. there's literally no need for the groups pkg_kubectl_cmd_util_CONSUMERS and pkg_kubectl_cmd_util_CONSUMERS_BAD if the bad deps are removed.  But they are still there, and this PR makes their nature less visible.

This PR says - hey, you don't need this map of the forest - the forest is right there, so we can remap it any time. In principal - yes.  But having put in the time, I'd prefer not to tear it up before the extraction is done.

Maybe in a distinct PR try to break a bad dep, and see if the rules are useful?  If not, then OK, we can tear it up.

Reply:
Thank you for your valuable feedback.

> but not a fan of the collapse.

Isn't it collapsing anyway? Every time a new kubectl package is created, someone must carefully add that new package to the forest, and we've clearly seen that this is NOT done.

BUT, Yes, I agree with you, it doesn't really matter if we call the forest a forest or a "a group of individual trees", we need to get rid of it one way or another.